pkg/models/datasource_cache.go

```diff
@@ -70,6 +67,26 @@ func (ds *DataSource) GetHttpTransport() (*http.Transport, error) {
 		IdleConnTimeout:       90 * time.Second,
 	}
 
+	ptc.cache[ds.Id] = cachedTransport{
+		Transport: transport,
+		updated:   ds.Updated,
+	}
+
+	return transport, nil
+}
+
+func (ds *DataSource) GetTLSConfig() (*tls.Config, error) {
```

Comment:
Decided to not add any cache around this for now since [sql engine](https://github.com/grafana/grafana/blob/master/pkg/tsdb/sql_engine.go#L86) has a cache for sql datasources.